samples/routing/flowcontrol/Core_6/Shared/FlowControl/FlowControl.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using NServiceBus;
+using NServiceBus.Features;
+using NServiceBus.Routing;
+using NServiceBus.Transports;
+
+public class FlowControl : Feature
+{
+    public FlowControl()
+    {
+        Defaults(s => s.Set<FlowManager>(new FlowManager()));
+    }
+
+    protected override void Setup(FeatureConfigurationContext context)
+    {
+        var sessionId = Guid.NewGuid().ToString();
+        var flowManager = context.Settings.Get<FlowManager>();
+        var controlAddress = context.Settings.InstanceSpecificQueue() ?? context.Settings.LocalAddress();
+
+        var instance = context.Settings.Get<TransportInfrastructure>().BindToLocalEndpoint(
+            new EndpointInstance(context.Settings.EndpointName(), context.Settings.GetOrDefault<string>("EndpointInstanceDiscriminator")));
```

Comment:
Is this going to be a common operation? Should there be an extension method to get the local `EndpointInstance` out of the settings?